XenonUtils/xex.cpp

```diff
@@ -192,6 +194,79 @@ Image Xex2LoadImage(const uint8_t* data, size_t dataSize)
                 destData += blocks[i].zeroSize;
             }
         }
+        else if (fileFormatInfo->compressionType == XEX_COMPRESSION_NORMAL)
+        {
+            result = std::make_unique<uint8_t[]>(imageSize);
+            auto* destData = result.get();
+
+            const Xex2CompressedBlockInfo* blocks = &((const Xex2FileNormalCompressionInfo*)(fileFormatInfo + 1))->firstBlock;
+            const uint32_t headerSize = header->headerSize.get();
+
+            const uint32_t exeLength = dataSize - headerSize;
+            const uint8_t* exeBuffer = srcData;
+
+            uint8_t* compressBuffer = NULL;
+            const uint8_t* p = NULL;
+            uint8_t* d = NULL;
+            sha1::SHA1 s;
+
+            compressBuffer = (uint8_t*)calloc(1, exeLength);
```

Comment:
You can use `std::unique_ptr<uint8_t[]>/std::make_unique<uint8_t[]>` for the allocation instead of allocating it manually.

Reply:
I've swapped it to std::make_unique<uint8_t[]>